### PR TITLE
Allows the user to request the number of comments displayed in the index page

### PR DESCRIPTION
**Pull request by Cayle**
Now, the user can modify the amount of comments that is been displayed.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -37,12 +37,22 @@
...
         DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
         PreparedQuery results = datastore.prepare(query);
+        int result_length =0;
+
+        for (Entity commentEntity : results.asIterable()) {
+            result_length++;
+        }
 
         List<Comments> comments = new ArrayList<>();
+        int no_of_comments = Integer.parseInt(request.getParameter("no"));
```
Comment (rayliu0317) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:54]: This isn't a very intuitive variable name. 'no' can mean 'No' or 'Number'.

I would call it comment_count_target

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -37,12 +37,22 @@
...
+        int count_comments = 0;
```
Comment (rayliu0317) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:55]: comment_count is a better name I think.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -37,12 +37,22 @@
 
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        System.out.println("hey, whats up !");
```
Comment (rayliu0317) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:40]: Do you want to keep these here or are these just for debugging purpose?

File: portfolio/src/main/webapp/index.html [hunk 4]
```diff
@@ -26,7 +26,7 @@
 
     <title>Caleb's Portfolio</title>
   </head>
-  <body class="body-image" onload="retrieveComments()">
+  <body class="body-image" onload="getCountComments()">
```
Comment (rayliu0317) [inline portfolio/src/main/webapp/index.html:29]: Same here getCommentCount is a better name

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -87,16 +87,39 @@ function changePattern(pattern) {
   }
 }
 
-async function retrieveComments() {
```
Comment (rayliu0317) [inline portfolio/src/main/webapp/script.js:90]: We don't normally encourages commenting out code actually. I think you can just delete it if you don't plan to use/keep it.

File: portfolio/src/main/webapp/script.js [hunk 6]
```diff
@@ -87,16 +87,39 @@ function changePattern(pattern) {
...
+//     message_json.forEach(function(comments) {
+//         commentHtml = commentHtml + comments.comment + " --- -- " + comments.name + "<br>";
+//     })
+
+//     commentContainer.innerHTML = commentHtml;
+// }
+
+async function getCountComments(){
+    var no_of_comment = document.getElementById("no_of_comments").value;
+    const response = await fetch("/data?no="+no_of_comment);
```
Comment (rayliu0317) [inline portfolio/src/main/webapp/script.js:113]: spaces between +

File: walkthroughs/week-3-server/examples/todo-list/src/main/java/com/google/sps/servlets/NewTaskServlet.java [hunk 7]
```diff
@@ -35,7 +35,7 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     Entity taskEntity = new Entity("Task");
     taskEntity.setProperty("title", title);
     taskEntity.setProperty("timestamp", timestamp);
-
+    `
```
Comment (rayliu0317) [inline walkthroughs/week-3-server/examples/todo-list/src/main/java/com/google/sps/servlets/NewTaskServlet.java:38]: this seems unintentional

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 8]
```diff
@@ -51,6 +61,10 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
             long timestamp = (long) commentEntity.getProperty("timestamp");
             Comments newComment =  new Comments(id, comment, name, timestamp);
             comments.add(newComment);
+            count_comments++;
+            if (count_comments >= no_of_comments ) {
```
Comment (rayliu0317) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:65]: remove extra space.

Comment (rayliu0317) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:65]: You can actually have "if (++count_comments >= comment_count) {" here and remove line 64.

Comment (Cayle) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: I have resolved all the other changes required.

File: walkthroughs/week-3-server/examples/todo-list/src/main/java/com/google/sps/servlets/NewTaskServlet.java (hunk 7 above)
Comment (Cayle) [reply walkthroughs/week-3-server/examples/todo-list/src/main/java/com/google/sps/servlets/NewTaskServlet.java:38]: Yes, it was. As at now, I can't seem to get the appropriate command to undo it.

**Pull request merged**